### PR TITLE
Refactor Config

### PR DESCRIPTION
**Pull request by haszi**
This PR refactors the `Config` class from using an array of options and method calls to access these to mostly using object properties. Probably best reviewed commit by commit.

### CODE REVIEW

File: phpdotnet/phd/Config.php [hunk 1]
```diff
@@ -5,49 +5,51 @@ class Config
...
-        'package_dirs'      => array(__INSTALLDIR__),
-        'saveconfig'        => false,
-        'quit'              => false,
-        'indexcache'        => '',
-        'memoryindex'       => '',
-    );
-
-    private static $optionArray = null;
+    /** @var array<string, string> */
+    public array $output_format = [];
```
Comment (Girgias) [inline phpdotnet/phd/Config.php:10]: Is the change from static to instance properties okay? (not that I think having it static in the first place made a lot of sense).

**Review by Girgias — APPROVED** [review]
MSTM, I think it would be better to use camelCase for the property names, but this can wait.

File: phpdotnet/phd/Config.php (hunk 1 above)
Comment (haszi) [reply phpdotnet/phd/Config.php:10]: I think it makes sense that one set of configuration options can be assigned to one object of the class Config and then we just pass the appropriate object when needed.

As far as how Config is used in PhD, this doesn't make much of a difference as only one Config object is used throughout. :-)

**Comment by haszi** [comment]
> MSTM, I think it would be better to use camelCase for the property names, but this can wait.

I wanted to do this eventually but it makes more sense to do this as part of this PR.

**Review by Girgias — APPROVED** [review]
Let me know if you are happy as if for me to merge it as a rebase or if you want to stilll organise the commits.

**Comment by haszi** [comment]
> Let me know if you are happy as if for me to merge it as a rebase or if you want to stilll organise the commits.

I'll leave it up to you to merge it as you want.

**Pull request merged**